app/domain/processor.js

```diff
@@ -1,33 +1,51 @@
 const { ContainerClient } = require('@azure/storage-blob') // eslint-disable-line no-unused-vars
 const crypto = require('crypto')
-const { uploadChunkToBlob, withTimeout } = require('../services/blob-client')
 const { chunkDocument } = require('../utils/chunker')
+const { uploadDocument, deleteDocuments } = require('../services/azure-search-service')
+const { SearchClient } = require('@azure/search-documents') // eslint-disable-line no-unused-vars
+const { generateEmbedding } = require('../services/openai-service')
 
 /**
- * @typedef {{link: string, lastModified: string}} Manifest
+ * @typedef {{link: string, lastModified: string, documentKeys: string[]}} Manifest
  */
 
 /**
- * Chunks and uploads grants to azure blob storage
- * @param {{title: string, content: string, updateDate: Date, url: string}[]} grants
+ * Chunks and uploads grants to AI Search & removes old grants from AI Search
+ * @param {import('../services/govuk-api').Grant[]} grants
  * @param {Manifest[]} manifestGrants
  * @param {string} schemeName
  * @param {ContainerClient} containerClient
+ * @param {SearchClient} searchClient
  * @returns {{number, processedGrants: Manifest[]}}
  */
-const processGrants = async (grants, manifestGrants, schemeName, containerClient) => {
-  let chunkCount = 0
+const process = async (grants, manifestGrants, schemeName, containerClient, searchClient) => {
```

Comment:
What do you think about wrapping params for the function in an object 
`{grants, manifestGrants, schemeName, containerClient, searchClient}`

when its more than 3 params, to ensure the js doesn't break randomly if a dev accidentally up the order of props  going into this funciton